sql/src/main/java/io/crate/planner/UnionFlatteningVisitor.java

```diff
@@ -0,0 +1,57 @@
+package io.crate.planner;
+
+import io.crate.analyze.MultiSourceSelect;
+import io.crate.analyze.QueriedSelectRelation;
+import io.crate.analyze.QueriedTable;
+import io.crate.analyze.TwoRelationsUnion;
+import io.crate.analyze.relations.AnalyzedRelationVisitor;
+import io.crate.analyze.relations.QueriedDocTable;
+import io.crate.analyze.relations.QueriedRelation;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Processes a tree of Unions and returns a flat List of all sub-relations
+ */
+public class UnionFlatteningVisitor
```

Comment:
I extracted it as public because it's needed in the UnionConsumer and SelectStatementPlanner